lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -1702,6 +1702,7 @@ public function get_collection_params() {
 		$params['filter'] = array(
 			'description'       => __( 'Use WP Query arguments to modify the response; private query vars require appropriate authorization.' ),
 		);
+		$params['filter[posts_per_page]'] = $params['per_page'];
```

Comment:
I believe most of the API would write that array like $params['filter']['posts_per_page'] would have to double check though.